### ISSUE TITLE
Function getimagesize can't find file after page was deleted

### ISSUE THREAD

**Issue by dennisspohr** [processwire/processwire-issues#356]
### Description of the issue

If I delete a page (`$pages->delete($page, true);`) with an image-field, I get the following error:
> getimagesize(D:/xampp/htdocs/.../site/assets/files/8188/image.png): failed to open stream: No such file or directory
> (D:\xampp\htdocs\...\wire\core\Pageimage.php on Line 239)

The file definitely exists before I deleted the page. After deleting the page the folder (8188) is gone, and it seems that PW still wants to access it.

### Setup/Environment
* ProcessWire version:  3.0.62

**Comment by TomS-** [comment]
I've come across this before, it should check if the directory exists before trying to delete.

**Comment by ryancramerdesign** [comment]
I'm not seeing this on regular page deletions, but am wondering if you mean literally deleting the page being viewed? (i.e. the $page API var?). If that's the case, then I can see how this could occur, so we might have to put in an exception to prevent the page represented by the $page API var from being deleted.

**Comment by dennisspohr** [comment]
Yes Ryan that's the case. I'm deleting the current page and redirect the user to another one showing a success-message. The delete-process actual works, just getting this error regarding the image-file.

Would be great to be able to delete the current page. 
I'm redirecting the logged-in user to this page /path/my-page?delete=1 to delete it.

I think it's a quite practical use-case.

**Comment by ryancramerdesign** [comment]
Deleting the page being viewed is something you want to avoid, I don't
think it's safe to do. There is an entire shutdown process that accompanies
every request (even after you've issued a redirect), and a working $page
variable is expected for the entire request. In addition, when modules go
through their destruct sequence, they may still be using the current $page
for maintenance and such. A safer thing to do would be to unpublish, hide
or maybe even trash the page. But either way, use POST for a request that
causes changes to the database, especially destructive ones.

**Comment by Toutouwai** [comment]
> I'm deleting the current page and redirect the user to another one showing a success-message.

Might be better to avoid the issue by deleting my-page via the page you are redirecting to, i.e. a URL like: /path/delete/?page=1234
You would check that the user is allowed to delete page with ID 1234 and if so show the success message.

**Comment by ryancramerdesign** [comment]
I don't think there's anything PW can do to make it okay for you to delete the current page being viewed, but I have added a check that will throw an Exception if you attempt to delete the current page, which should make it more clear. Just in case anyone has already implemented their own workarounds for this, I've made the exception only apply for new installations after today's date. Thanks.
